google-cloud-storage/src/main/java/com/google/cloud/storage/spi/v1/HttpStorageRpc.java

```diff
@@ -704,6 +705,10 @@ public long read(
       StorageException serviceException = translate(ex);
       if (serviceException.getCode() == SC_REQUESTED_RANGE_NOT_SATISFIABLE) {
         return Tuple.of(null, new byte[0]);
+      } else if (serviceException.getMessage().contains("Connection closed prematurely")) {
+        serviceException = new StorageException(new SocketException(serviceException.getMessage()));
```

Comment:
This is very weird. I don't think I've ever seen this double wrapping in one throw clause done before. Can you explain the purpose here and how it works?